flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -0,0 +1,291 @@
+
+		startCommandValues.put("class", mainClass.getName());
+
+		startCommandValues.put("args", mainArgs != null ? mainArgs : "");
+
+		startCommandValues.put("redirects",
+			"1> " + logDirectory + "/taskmanager.out " +
+			"2> " + logDirectory + "/taskmanager.err");
+
+		final String commandTemplate = flinkConfig
+			.getString(KubernetesConfigOptions.CONTAINER_START_COMMAND_TEMPLATE);
+		return BootstrapTools.getStartCommand(commandTemplate, startCommandValues);
+	}
+
+	/**
+	 * Get config map volume for job manager and task manager pod.
+	 * @param clusterId Cluster id.
+	 * @param hasLogback Uses logback?
+	 * @param hasLog4j Uses log4j?
+	 * @return Config map volume.
+	 */
+	public static Volume getConfigMapVolume(
+		String clusterId,
+		boolean hasLogback,
+		boolean hasLog4j) {
+
+		Volume configMapVolume = new Volume();
+		configMapVolume.setName(FLINK_CONF_VOLUME);
+		List<KeyToPath> items = new ArrayList<>();
+		items.add(new KeyToPath(FLINK_CONF_FILENAME, null, FLINK_CONF_FILENAME));
+
+		if (hasLogback) {
+			items.add(new KeyToPath(CONFIG_FILE_LOGBACK_NAME, null, CONFIG_FILE_LOGBACK_NAME));
+		}
+
+		if (hasLog4j) {
+			items.add(new KeyToPath(CONFIG_FILE_LOG4J_NAME, null, CONFIG_FILE_LOG4J_NAME));
+		}
+		configMapVolume.setConfigMap(new ConfigMapVolumeSourceBuilder()
+			.withName(CONFIG_MAP_PREFIX + clusterId)
+			.withItems(items)
+			.build());
+		return configMapVolume;
+	}
+
+	/**
+	 * Get config map volume for job manager and task manager pod.
+	 * @param flinkConfDirInPod Flink conf directory that will be mounted in the pod.
+	 * @param hasLogback Uses logback?
+	 * @param hasLog4j Uses log4j?
+	 * @return Volume mount list.
+	 */
+	public static List<VolumeMount> getConfigMapVolumeMount(
+		String flinkConfDirInPod,
+		boolean hasLogback,
+		boolean hasLog4j) {
+
+		List<VolumeMount> volumeMounts = new ArrayList<>();
+		volumeMounts.add(new VolumeMountBuilder()
+			.withName(FLINK_CONF_VOLUME)
+			.withMountPath(new File(flinkConfDirInPod, FLINK_CONF_FILENAME).getPath())
+			.withSubPath(FLINK_CONF_FILENAME).build());
+
+		if (hasLogback) {
+			volumeMounts.add(new VolumeMountBuilder()
+				.withName(FLINK_CONF_VOLUME)
+				.withMountPath(new File(flinkConfDirInPod, CONFIG_FILE_LOGBACK_NAME).getPath())
+				.withSubPath(CONFIG_FILE_LOGBACK_NAME)
+				.build());
+		}
+
+		if (hasLog4j) {
+			volumeMounts.add(new VolumeMountBuilder()
+				.withName(FLINK_CONF_VOLUME)
+				.withMountPath(new File(flinkConfDirInPod, CONFIG_FILE_LOG4J_NAME).getPath())
+				.withSubPath(CONFIG_FILE_LOG4J_NAME)
+				.build());
+		}
+		return volumeMounts;
+	}
+
+	/**
+	 * Get resource requirements from memory and cpu.
+	 * @param mem Memory in mb.
+	 * @param cpu cpu.
+	 * @return Resource requirements.
+	 */
+	public static ResourceRequirements getResourceRequirements(int mem, double cpu) {
+
+		Quantity cpuQuantity = new Quantity(String.valueOf(cpu));
+		Quantity memQuantity = new Quantity(String.valueOf(mem + Constants.RESOURCE_UNIT_MB));
```

Comment:
unnecessary `String.valueOf`

Reply:
I will remove the `String.valueOf` for memQuantity.